### DISCUSSION TITLE
Need help : Dev server unable to hot reload upon file change in docker volume

### DISCUSSION THREAD

**Discussion by gaetleba** [remix-run/remix#6769] [Q&A]
Hello everyone. 
My remix dev server does not rebuild or reload upon file changes. Here is my setup:

- On windows 10 with docker desktop (4.21.1 with docker engine 24.0.2)
- Docker-compose with a volume like so:
```  app:
    build:
      context: ./app
      dockerfile: ./Dockerfile
    volumes:
      - ./app/app:/code/app
```
- app folder with remix project and Dockerfile like so:
```
FROM node:alpine
WORKDIR /code
COPY . .
RUN npm i
CMD ["npm", "run", "dev"]
```
- which is setup with:
  - Remix 1.18
  - Basic default app
  - v2_dev enabled
- an API container
- a nginx container with proxies to the API and the remix container

When I do ``docker-compose up --build`` everything goes well, upon opening the remix frontend in my browser the app works and dev websocket connection is correctly initialized. 

However when I change a file in my remix/app folder nothing happens. The remix dev server does not rebuild or reload. If I open the container CLI and check the file on disk it is updated though.

If I'm out of docker and simply run ``npm run dev`` it works as expected of course.

To me it looks similar to https://stackoverflow.com/questions/76296335/remix-dev-doesnt-detect-file-changes-inside-a-docker-container-with-volumes-con, except the exact volume mapping does not improve the situation in my case and I don't know what they mean by "using nodemon"

**Comment by gaetleba** [comment]
I fixed it by adding 
```
    environment:
      - CHOKIDAR_USEPOLLING=true
```
to the docker-compose file.

**Reply by rameshdon1** [reply]
@gaetleba thanks that worked for me. I added that env to docker compose file and worked.
